src/psyclone/psyir/transformations/parallel_loop_trans.py

```diff
@@ -406,3 +413,6 @@ def apply(self, node, options=None):
 
         # Add the loop directive as a child of the node's parent
         node_parent.addchild(directive, index=node_position)
+
+        if nowait:
+            self._add_asynchronicity(node, directive)
```

Comment:
Thinking about this some more, what will happen if a user specifies nowait=True for ACCLoopTrans()? Should we catch that situation (as it's not yet supported)?

Reply:
Nothing would happen, as the default `_add_asynchronicity` function does nothing. Since it causes no issues with correctness, and we can never guarantee to add asynchronicity as part of the transformation my feeling is nothing is fine.

I can add a comment to the base class `_add_asynchronicity` to TODO #11 to log if nothing is happening though.